src/wheel/cli/tags.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import annotations
+
+import itertools
+import os
+
+from ..wheelfile import WheelFile
+from .pack import read_tags, set_build_number
+
+
+def compute_tags(original_tags: list[str], new_tags: list[str] | None) -> list[str]:
+    """Add or replace tags."""
+
+    if not new_tags:
+        return original_tags
+
+    if new_tags[0] == "":
+        return original_tags + new_tags[1:]
+    else:
+        return new_tags
+
+
+def tags(
+    wheel: str,
+    python_tags: list[str] | None = None,
+    abi_tags: list[str] | None = None,
+    platform_tags: list[str] | None = None,
+    build_number: int | None = None,
+    remove: bool = False,
+) -> str:
+    """Change the tags on a wheel file.
+
+    The tags are left unchanged if they are not specified. To specify "none",
+    use ["none"]. To append to the previous tags, use ["", ...].
+
+    :param wheels: The paths to the wheels.
+    :param python_tags: The Python tags to set.
+    :param abi_tags: The ABI tags to set.
+    :param platform_tags: The platform tags to set.
+    :param build_number: The build number to set.
+    :param remove: Remove the original wheel.
+    """
+    with WheelFile(wheel, "r") as f:
+        wheel_info = f.read(f.dist_info_path + "/WHEEL")
+
+        original_wheel_name = os.path.basename(f.filename)
+        namever = f.parsed_filename.group("namever")
+        build = f.parsed_filename.group("build")
+        original_python_tags = f.parsed_filename.group("pyver").split(".")
+        original_abi_tags = f.parsed_filename.group("abi").split(".")
+        orignial_plat_tags = f.parsed_filename.group("plat").split(".")
+
+    tags, existing_build_number = read_tags(wheel_info)
+
+    impls = {tag.split("-")[0] for tag in tags}
+    abivers = {tag.split("-")[1] for tag in tags}
+    platforms = {tag.split("-")[2] for tag in tags}
+
+    if impls != set(original_python_tags):
```

Comment:
Hmm, I'd usually say asserts should only be for things known to be true or internal consistency (since they can be turned off). But this one is tricky - I still think it's serious enough if this is broken, it would be best to report a failure even if asserts were off. Maybe there's a better error class to throw?